Site/admin/components/Account/MergeSummary.php

```diff
@@ -0,0 +1,337 @@
+				$this->app->relocate(sprintf(
+					'Account/MergeConfirm?id=%s&id2=%s&keep_first=1',
+					$this->id,
+					$this->id2
+				));
+
+			} else if ($this->ui->getWidget('keep_second_button')->hasBeenClicked()) {
+				$this->app->relocate(sprintf(
+					'Account/MergeConfirm?id=%s&id2=%s&keep_first=0',
+					$this->id,
+					$this->id2
+				));
+			}
+		}
+	}
+
+	// }}}
+	// build phase
+	// {{{ protected function buildInternal()
+
+	protected function buildInternal()
+	{
+		parent::buildInternal();
+
+		$this->buildAccountDetailsFrame();
+	}
+
+	// }}}
+	// {{{ protected function getAccountDetailsStore()
+
+	protected function getAccountDetailsStore($account)
+	{
+		$ds = new SwatDetailsStore($account);
+		$ds->fullname = $account->getFullname();
+		return $ds;
+	}
+
+	// }}}
+	// {{{ protected function getAccount()
+
+	protected function getAccount($id, $account)
+	{
+		if ($account === null) {
+			$account_class = SwatDBClassMap::get('SiteAccount');
+
+			$account = new $account_class();
+			$account->setDatabase($this->app->db);
+
+			if (!$account->load($id)) {
+				throw new AdminNotFoundException(sprintf(
+					Site::_('An account with an id of ‘%d’ does not exist.'),
+					$id
+				));
+			}
+
+			$instance_id = $this->app->getInstanceId();
+			if ($instance_id !== null) {
+				if ($account->instance->id !== $instance_id) {
+					throw new AdminNotFoundException(sprintf(
+						Store::_('Incorrect instance for account ‘%d’.'),
+						$id
+					));
+				}
+			}
+		}
+
+		return $account;
+	}
+
+	// }}}
+	// {{{ protected function buildAccountDetails()
+
+	protected function buildAccountDetailsFrame()
+	{
+		$ds1 = $this->getAccountDetailsStore($this->account1);
+		$ds2 = $this->getAccountDetailsStore($this->account2);
+
+
+		$this->buildAccountDetails(
+			$ds1,
+			$this->ui->getWidget('details_view_left')
+		);
+
+		$this->buildAccountDetails(
+			$ds2,
+			$this->ui->getWidget('details_view_right')
+		);
+
+		$details_frame = $this->ui->getWidget('details_frame');
+		$details_frame->title = Site::_('Merge Accounts');
+		$details_frame->subtitle = sprintf(
+			'%s, %s',
+			$ds1->fullname,
+			$ds2->fullname
+		);
+	}
+
+	// }}}
+	// {{{ protected function buildAccountDetails()
+
+	protected function buildAccountDetails($ds, $details_view)
+	{
+		$date_field = $details_view->getField('createdate');
+		$date_renderer = $date_field->getRendererByPosition();
+		$date_renderer->display_time_zone = $this->app->default_time_zone;
+
+		$details_view->data = $ds;
+	}
+
+	// }}}
+	// {{{ protected function buildNavBar()
+
+	protected function buildNavBar()
+	{
+		$this->navbar->addEntry(new SwatNavBarEntry(
+			$this->account1->fullname,
+			sprintf('Account/Details?id=%s', $this->id)
+		));
+
+		$this->navbar->addEntry(new SwatNavBarEntry(
+			Site::_('Merge'),
+			sprintf('Account/Merge?id=%s', $this->id)
+		));
+
+		$this->navbar->addEntry(new SwatNavBarEntry(sprintf(
+			Site::_('Merge With %s'),
+			$this->account2->fullname
```

Comment:
always use account->getFullname() so it works on sites where the first/last are split.